File: Baekjoon/Level5/5-01/main.go
package main

import (
    "bufio"
    "fmt"
    "os"
)

func main() {
    var n, nums int
    fmt.Scan(&n)

    r := bufio.NewReader(os.Stdin)
    a := make([]int, n)

    for i := 0; i < n; i++ {
        fmt.Fscan(r, &nums) // 대량의 입력 필요시 (빠르다)
        a[i] = nums
    }

    min, max := a[0], a[0]
    for _, num := range a {
        if min > num {
            min = num
        }
        if max < num {
            max = num
        }
    }
    fmt.Println(min, max)
}

File: Baekjoon/Level5/5-01/main.dart
import 'dart:io';

String solution(int n, List<int> numbers) {
  String answer = '';
  int min = numbers[0];
  int max = numbers[0];
  for (var n in numbers) {
    if (min > n) min = n;
    if (max < n) max = n;
  }
  answer += '$min $max';
  return answer;
}

void main() {
  int n = int.parse(stdin.readLineSync()!);
  List<String> sc = stdin.readLineSync()!.split(' ');
  var numbers = <int>[];
  for (int i = 0; i < n; i++) {
    numbers.add(int.parse(sc[i]));
  }

  print(solution(n, numbers));
}
